pkg/handler/data_recorder_pubsub.go

```diff
@@ -0,0 +1,89 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/checkr/flagr/pkg/config"
+	"github.com/checkr/flagr/swagger_gen/models"
+	"google.golang.org/api/option"
+
+	"github.com/sirupsen/logrus"
+)
+
+type pubsubRecorder struct {
+	enabled  bool
+	producer *pubsub.Client
+	topic    *pubsub.Topic
+}
+
+// NewPubsubRecorder creates a new Pubsub recorder
+var NewPubsubRecorder = func() DataRecorder {
+	client, err := pubsub.NewClient(
+		context.Background(),
+		config.Config.RecorderPubsubProjectID,
+		option.WithCredentialsFile(config.Config.RecorderPubsubKeyFile),
+	)
+	if err != nil {
+		logrus.WithField("pubsub_error", err).Error("error getting pubsub client")
```

Comment:
let's `Fatal` here instead of `Error`

Reply:
I think the issue is caused by `get` method from pubsub which is a blocking operation, see pubsub doc [here](https://godoc.org/cloud.google.com/go/pubsub#PublishResult.Get)
we should pass context either with a deadline or timeout to `get` method  instead of just background context.

Reply:
Can confirm this is the issue. When I disable verbose logging, everything works as expected.